src/Exception/InvalidOrderStateException.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+namespace PrestaShop\Module\PrestashopCheckout\Exception;
+
+use Throwable;
+
+/**
+ * Thrown when the order state is incompatible with an action (ex: standard
+ * refund on an order not paid yet).
+ */
+class InvalidOrderStateException extends \Exception
+{
+    /**
+     * Used when the order is not paid (and it should be)
+     */
+    const NOT_PAID = 1;
+
+    /**
+     * Used when the order is already paid (and it should not be)
+     */
+    const ALREADY_PAID = 2;
+
+    /**
+     * Used when the order has not been delivered (and it should have)
+     */
+    const DELIVERY_NOT_FOUND = 3;
+
+    /**
+     * Used when the order has been delivered (and it shouldn't have)
+     */
+    const UNEXPECTED_DELIVERY = 4;
+    /**
+     * Used when the order state is not found
+     */
+    const INVALID_ID = 5;
+
+    /**
+     * @param int $code
+     * @param string $message
+     * @param Throwable|null $previous
+     */
+    public function __construct($code = 0, $message = '', Throwable $previous = null)
```

Comment:
```suggestion
    public function __construct($code = 0, $message = '', $previous = null)
```
Not PHP 5.6 compliant 😞